src/main.zig

```diff
@@ -381,6 +383,57 @@ pub fn Deserializer(comptime ReaderType: type) type {
             };
         }
 
+        /// Deserializes an enum (stored as an integer)
+        pub fn deserializeEnum(self: *Self, comptime T: type) ReadError!T {
+            if (@typeInfo(T) != .Enum) @compileError("Expected enum type, but found type '" ++ @typeName(T) ++ "'");
+
+            const I = comptime meta.Tag(T);
```

Comment:
We should verify it's a tagged enum. i.e. `enum(u8)`. Right now the type is inferred by the given amount of fields the enum contains. This means that removing a field, or adding a field could change its type. i.e. from u8 to u9. This would implicitly change the encoding of the type which the user may not notice. Enforcing the enum to be tagged, the change is a lot more explicit as you get a nice compilation error if the tag type would change.

Reply:
It seems you're right, there isn't a way to verify this. In which case I'm okay with the user of the library being responsible for documenting it correctly.